src/nomad_simulations/physical_property.py

```diff
@@ -0,0 +1,231 @@
+        """,
+    )
+
+    physical_property_ref = Quantity(
+        type=Reference(SectionProxy('PhysicalProperty')),
+        description="""
+        Reference to the `PhysicalProperty` section from which the physical property was derived. If `physical_property_ref`
+        is populated, the quantity `is_derived` is set to True via normalization.
+        """,
+    )
+
+    is_derived = Quantity(
+        type=bool,
+        default=False,
+        description="""
+        Flag indicating whether the physical property is derived from other physical properties. We make
+        the distinction between directly parsed and derived physical properties:
+            - Directly parsed: the physical property is directly parsed from the simulation output files.
+            - Derived: the physical property is derived from other physical properties. No extra numerical settings
+                are required to calculate the physical property.
+        """,
+    )
+
+    is_scf_converged = Quantity(
+        type=bool,
+        description="""
+        Flag indicating whether the physical property is converged or not after a SCF process. This quantity is connected
+        with `SelfConsistency` defined in the `numerical_settings.py` module.
+        """,
+    )
+
+    self_consistency_ref = Quantity(
+        type=SelfConsistency,
+        description="""
+        Reference to the `SelfConsistency` section that defines the numerical settings to converge the
+        physical property.
+        """,
+    )
+
+    @property
+    def variables_shape(self) -> Optional[list]:
+        """
+        Shape of the variables over which the physical property varies. This is extracted from
+        `Variables.n_grid_points` and appended in a list.
+
+        Example, a physical property which varies with `Temperature` and `ElectricField` will
+        return `variables_shape = [n_temperatures, n_electric_fields]`.
+
+        Returns:
+            (list): The shape of the variables over which the physical property varies.
+        """
+        if self.variables is not None:
+            return [v.get_n_grid_points(v.grid_points, logger) for v in self.variables]
+        return []
+
+    @property
+    def full_shape(self) -> list:
+        """
+        Full shape of the physical property. This quantity is calculated as:
+            `full_shape = variables_shape + shape`
+        where `shape` is passed as an attribute of the `PhysicalProperty` and is related with the order of
+        the tensor of `value`, and `variables_shape` is obtained from `variables_shape` and is
+        related with the shapes of the `variables` over which the physical property varies.
+
+        Example: a physical property which is a 3D vector and varies with `variables=[Temperature, ElectricField]`
+        will have `shape = [3]`, `variables_shape=[n_temperatures, n_electric_fields]`, and thus
+        `full_shape=[n_temperatures, n_electric_fields, 3]`.
```

Comment:
Now _property_ will be responsible for containing its own definition along with dependent variables.
This leads to the awkward case where extracting `Energy` data, I have to plow through all its variables...
So, every time a data scientist wants to manipulate any `PhysicalPropertyBaseclass`, they first have to access `full_shape` and `variable_shape` to extract the underlying property itself...

How about `PhysicalPropertyBaseclass` actually stores `value` and `dependent_variables` separately, and if the user wants the manifold, they access both?

Reply:
You also commented:
> Btw, why can't these be other properties in their own right?

I think it is hard for me to understand specific cases where this might happen. I am guessing you are thinking about physical properties that are **functionals**, but for those, I think anyways we should see some example before taking a decision.

In any other case, I'll argue that the concept of `Energy(Variables)` is slightly different from `Energy(PhysicalProperty)`: the first is just the discretization of your space for measuring a property, the later is the property you are measuring directly. I am pretty sure Area A will disagree with this, but I will argue that in any case, when someone is doing a measurement of, e.g., an spectra, you **vary** a certain parameter while **measuring** what is happening to the other.

___

Do you have in any case some specific example where this does not happen like I am arguing?